### ISSUE TITLE
rds-deployment-existing-ad * DSC failed * Error downloading https://raw.githubusercontent.com/Azure/azure-QuickStart-Templates/master/rds-deployment//Configuration.zip

### ISSUE THREAD

**Issue by rstamp** [Azure/azure-quickstart-templates#5486]
rds-deployment-existing-ad https://github.com/Azure/azure-quickstart-templates/tree/master/rds-deployment-existing-ad/

### Issue Details
"message": "VM has reported a failure when processing extension 'sessionhost'. Error message: \"The DSC Extension failed to execute: Error downloading https://raw.githubusercontent.com/Azure/azure-QuickStart-Templates/master/rds-deployment//Configuration.zip after 29 attempts: The remote name could not be resolved: 'raw.githubusercontent.com'.

### Repro steps (*if necessary, delete otherwise*)
1.
2.
3.
4.
5.

**Comment by pchalabi** [comment]
@rstamp I was having this same issue, it looks like the URL has an extra / before Configuration.zip.  I edited this but am still getting this same message, unfortunately.

**Comment by GrooveCS** [comment]
I'm having the same issue... Any provide a work-around yet?

**Comment by tapasudenia** [comment]
{"code":"DeploymentFailed","message":"At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/arm-debug for usage details.","details":[{"code":"Conflict","message":"{\r\n \"status\": \"Failed\",\r\n \"error\": {\r\n \"code\": \"ResourceDeploymentFailure\",\r\n \"message\": \"The resource operation completed with terminal provisioning state 'Failed'.\",\r\n \"details\": [\r\n {\r\n \"code\": \"VMExtensionProvisioningError\",\r\n \"message\": \"VM has reported a failure when processing extension 'CreateADForest'. Error message: \\\"The DSC Extension failed to execute: Error downloading https://mystorageaccount.blob.core.windows.net/template-drop/0.1.0-mybranchName.1+134/ad-forest/DSC/CreateADRootDC1.ps1.zip after 29 attempts: The remote name could not be resolved: 'mystorageaccount.blob.core.windows.net'.\\nMore information about the failure can be found in the logs located under 'C:\\\\WindowsAzure\\\\Logs\\\\Plugins\\\\Microsoft.Powershell.DSC\\\\2.77.0.0' on the VM.\\\".\"\r\n }\r\n ]\r\n }\r\n}"}]}

I am getting the same error, any workaround or clue?

**Comment by dasutton** [comment]
Anyone have a work around for this problem?  Getting this very same problem this week concerning the double // before Configuration.zip.

**Comment by dasutton** [comment]
Thanks, but the sections of code that do not work for me are:

"variables": {
                "gwdnsLabelPrefix": "[toLower(concat('gwd', resourceGroup().name))]",
                "cbdnsLabelPrefix": "[toLower(concat('cbd', resourceGroup().name))]",
                "imagePublisher": "MicrosoftWindowsServer",
                "imageOffer": "WindowsServer",
                "dnsServerPrivateIp": "10.0.0.4",
                "subnet-id": "[concat(resourceId(parameters('adVnetRG'),'Microsoft.Network/virtualNetworks',parameters('adVnetName')),'/subnets/',parameters('adSubnetName'))]",
                "publicIpRef": "publicIp",
                "brokerIpRef": "brokerpublicIp",
                "gatewayIpRef": "gatewaypublicIp",
                "assetLocation": "https://raw.githubusercontent.com/Azure/azure-QuickStart-Templates/master/rds-deployment/"



And












"resources": [
                        {
                            "apiVersion": "2015-06-15",
                            "type": "Microsoft.Compute/virtualMachines/extensions",
                            "name": "gw-vm/gateway",
                            "location": "[parameters('location')]",
                            "dependsOn": [
                                "[resourceId('Microsoft.Compute/virtualMachines', 'gw-vm')]"
                            ],
                            "properties": {
                                "publisher": "Microsoft.Powershell",
                                "type": "DSC",
                                "typeHandlerVersion": "2.11",
                                "autoUpgradeMinorVersion": true,
                                "settings": {
                                    "ModulesUrl": "[concat(variables('assetLocation'),'/Configuration.zip')]",
                                    "ConfigurationFunction": "Configuration.ps1\\Gateway",

In the first code section, note assetLocation path.  After the path there is a "/" and in the section code section before the zip file there is a "/"
This produces https://raw.githubusercontent.com/Azure/azure-QuickStart-Templates/master/rds-deployment//Configuration.zip when everything runs.

I see some differences in yours, such as the DSC/filename.ps1.zip mention.  I just don't know where your template is getting the full path to the file.

________________________________

**Comment by tapasudenia** [comment]
So my template gets the path from the Azure Storage account BLOB. And this template gets invoked from the Azure DevOps CI/CD Pipeline.
So remove the '/' if you don't have any other inner directory. So my path is template/ad-forest/DSC/CreateADRootDC1.ps1.zip

**Comment by dasutton** [comment]
Removing the extra "/" still produces the same error message.  Changing the assetLocation path from raw.gitbhubusercontent.com to just GitHub.com also produces the same message.  I also tried the IP address.  Remote name could not be resolved.

I am going to try and load the Configuration.zip file in my Azure blob storage and reference that path versus GitHub.

________________________________

**Comment by tapasudenia** [comment]
Yes, that should definitely work, because in my case, the code below
[concat(uri(deployment().properties.templateLink.uri,'DSC/CreateADRootDC1.ps1.zip'),parameters('mySasToken'))]

is fetching it from the Azure BLOB. I am just using template link to fetch it from the folder structure of my BLOB.

**Comment by dasutton** [comment]
Currently getting this now:

Managed disk storage account type change through Virtual Machine 'gw-vm' is not allowed. Please update disk resource at /subscriptions/***-***-**-**-*****/resourceGroups/****/providers/Microsoft.Compute/disks/gw_OSDisk.\",\r\n    \"target\": \"osDisk.managedDisk.storageAccountType

Thought it was talking about Standard HDD versus Standard SSD.  Not the case.

Thoughts?

________________________________

**Comment by tapasudenia** [comment]
Can you paste your Virtual Machine 'gw-vm' json? And do you have a separate json for Storage account in your template?

**Comment by dasutton** [comment]
I figured that one out.  It didn't like that the resource was already in place.  Apparently the script (which I did not write) does not account for that. I was trying to redeploy after another error.  

I am stuck though.  Getting:

"code": "Conflict",
      "message": "{\r\n  \"status\": \"Failed\",\r\n  \"error\": {\r\n    \"code\": \"ResourceDeploymentFailure\",\r\n    \"message\": \"The resource operation completed with terminal provisioning state 'Failed'.\",\r\n    \"details\": [\r\n      {\r\n        \"code\": \"VMExtensionProvisioningError\",\r\n        \"message\": \"VM has reported a failure when processing extension 'gateway'. Error message: \\\"The DSC Extension failed to execute: Error downloading [https://raw.githubusercontent.com/Azure/azure-QuickStart-Templates/master/rds-deployment/Configuration.zip]https://raw.githubusercontent.com/Azure/azure-QuickStart-Templates/master/rds-deployment/Configuration.zip after 29 attempts: The remote name could not be resolved: 'raw.githubusercontent.com'.\\n

Regardless of where the assetLocation path is set to (either GitHub or Azure Blob) getting this "The remote name could not be resolved:"

Original JSON :  https://github.com/Azure/azure-quickstart-templates/blob/master/rds-deployment-existing-ad/azuredeploy.json

For my copy in my repo I only changed the assetLocation to point to my storage account path in Azure.  I can do a NSLookup to my storage path.  Same error message as above though.

**Comment by robdyke** [comment]
@dasutton @rstamp see my reply to #6094 

You need to edit `        "dnsServerPrivateIp": "10.0.0.4",` in [azuredeploy.json line 105](https://github.com/Azure/azure-quickstart-templates/blob/cffab0d0e37657521ec0e30d79ea5c1740e413fc/rds-deployment-existing-ad/azuredeploy.json#L105).
